google/resource_kms_crypto_key.go

```diff
@@ -111,6 +112,46 @@ func resourceKmsCryptoKeyCreate(d *schema.ResourceData, meta interface{}) error
 	return resourceKmsCryptoKeyRead(d, meta)
 }
 
+func resourceKmsCryptoKeyUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	cryptoKeyId, err := parseKmsCryptoKeyId(d.Id(), config)
+	if err != nil {
+		return err
+	}
+
+	key := cloudkms.CryptoKey{Purpose: "ENCRYPT_DECRYPT"}
+
+	if d.HasChange("rotation_period") && d.Get("rotation_period") != "" {
```

Comment:
In theory, `d.Get("rotation_period")` will never be an empty string due to validation.
So if we set `rotation_period = ""`, it will fail validation, but that is the `master` behaviour. If we remove it completely, this block will not execute, but since we included an update mask, it will use the zero value of the struct, so it will end up setting the rotation period to `Never`. From my understanding, that would've been the case if it was not defined on `Create`.

Reply:
Oh, you're totally right- I missed the update mask. Thanks!